raiden/network/rpc/client.py

```diff
@@ -631,7 +631,10 @@ def __init__(self, jsonrpc_client, registry_address, startgas=GAS_LIMIT,
 
     def manager_address_by_token(self, token_address):
         """ Return the channel manager address for the given token. """
-        return self.proxy.channelManagerByToken.call(token_address)
+        manager_address_encoded = self.proxy.channelManagerByToken.call(token_address)
+        if not manager_address_encoded or manager_address_encoded == '0x':
+            manager_address_encoded = self.add_token(token_address)
```

Comment:
Hmm, I thought that was the idea.. I don't know a lot about the architecture, but this solution came in talks with @LefterisJP and @czepluch . If the expected the token would be already registered, this whole PR is pointless, and it should be rejected (including the test), and I can make some changes just to clarify this new expected behavior (`None` if token isn't registered, maybe some logging), and a new PR adding an endpoint for registering new tokens.